### ISSUE TITLE
NatRailPlugin: Stop Crash for Longer Departures

### ISSUE THREAD

**Issue by artemis-beta** [LanceMaverick/skybeard-2#57]
Example "Clapham Junction" returns too many entries

**Comment by nasfarley88** [comment]
New generator added to `utils` by @LanceMaverick might be useful: https://github.com/LanceMaverick/skybeard-2/blob/skb-dev/skybeard/utils.py#L71

**Comment by artemis-beta** [comment]
Has this been automatically implemented or do I need to call it within the plugin too?

**Comment by LanceMaverick** [comment]
You'll need to import it and use it like so
```Python
from skybeard.utils import partition_text


text = 'a very long string \n with line breaks \n that needs splitting'
messages = partition_text(text)
```
and messages will be a list of strings small enough to send. They will be split at the nearest line break to the character limit. You can then loop over the messages and send them. If the original `text` is small enough, it will return a list with one element (the original text string), so the sending loop with still work fine.

**Comment by LanceMaverick** [comment]
Closed as this is contained withing #74
